### PR TITLE
kafka role

### PR DESCRIPTION
**Pull request by ryane**
- [x] Installs cleanly on a fresh build of most recent master branch
- [x] Upgrades cleanly from the most recent release
- [x] Updates documentation relevant to the changes

Installation documentation available in the role readme. Note that you will need at least 3 workers each with 1 CPU and 4 GB of memory available to Mesos in order to run with the default configuration.

This depends on https://github.com/CiscoCloud/nginx-mantlui/pull/18 being merged and the 0.6.8 tag of ciscocloud/nginx-mantlui being pushed to docker hub.

After successful Kafka installation (it can take a few minutes), verify the following:

- that the mantl/kafka scheduler is running and healthy in Marathon
- that there are 3 Kafka brokers running in Mesos (broker-1, broker-2, broker-3)

Once the brokers are up, you should be able to produce and consume data. Here is one example of a manual test:

```shell
# create a topic
curl -sku admin:password -d "topic=journal" https://control-node-01/kafka/api/topic/add

# on one node, pipe journald logs into the journal kafka topic
sudo journalctl -f | sudo docker run --rm -i ryane/kafkacat -b broker-1.service.consul:9092,broker-2.service.consul:9092,broker-3.service.consul:9092 -t journal -z snappy

# on another node, consume the events from the  journal topic
sudo docker run --rm -it ryane/kafkacat -b broker-1.service.consul:9092,broker-2.service.consul:9092,broker-3.service.consul:9092 -t journal
```

### CODE REVIEW

File: roles/kafka/README.rst [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+   default: 512
+
+.. data:: kafka_broker_count
+
+   The number of Kafka brokers to start.
+
+   default: 3
+
+.. data:: kafka_broker_cpu
```
Comment (tymofii-polekhin) [inline roles/kafka/README.rst]: too many CPUs for only 4GB ram,
lets follow the 1/4 proportion (1cpu/4gb, 2cpu/8gb, ...)

**Comment by siddharthist** [comment]
We can see if the tag has been pushed on the [nginx-mantlui docker hub page](https://hub.docker.com/r/ciscocloud/nginx-mantlui/tags/).

File: roles/kafka/README.rst (hunk 1 above)
Comment (ryane) [inline roles/kafka/README.rst]: thanks, I updated the config as suggested

**Comment by ryane** [comment]
0.6.8 tag of the nginx-mantlui image has been published

File: roles/kafka/README.rst [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+   The port to bind to for the Kafka brokers.
+
+   default: 9092
+
+.. data:: kafka_broker_options
+
+   The Kafka options to pass to the brokers.
+
+   default: "log.flush.interval.ms=10000,num.recovery.threads.per.data.dir=1,delete.topic.enable=true,log.index.size.max.bytes=10485760,num.partitions=8,num.network.threads=3,socket.request.max.bytes=104857600,log.segment.bytes=536870912,log.cleaner.enable=true,zookeeper.connection.timeout.ms=1000000,log.flush.scheduler.interval.ms=2000,log.retention.hours=72,log.flush.interval.messages=20000,log.dirs=/mantl/a/dfs-data/kafka-logs\\,/mantl/b/dfs-data/kafka-logs\\,/mantl/c/dfs-data/kafka-logs\\,/mantl/d/dfs-data/kafka-logs\\,/mantl/e/dfs-data/kafka-logs\\,/mantl/f/dfs-data/kafka-logs,log.index.interval.bytes=4096,socket.receive.buffer.bytes=1048576,min.insync.replicas=2,replica.lag.max.messages=10000000,replica.lag.time.max.ms=1000000,log.retention.check.interval.ms=3600000,message.max.bytes=20480,default.replication.factor=2,zookeeper.session.timeout.ms=500000,num.io.threads=8,auto.create.topics.enable=false,socket.send.buffer.bytes=1048576"
```
Comment (siddharthist) [inline roles/kafka/README.rst]: I wonder if this could be broken up into a dict to be joined by templating, i.e.
```
kafka_broker_options:
  log.flush.interval.ms: "10000"
  num.recovery.threads.per.data.dir: "1"
  ...
```

Comment (stevendborrelli) [inline roles/kafka/README.rst]: I agree with this, it is hard to parse this command

**Comment by stevendborrelli** [comment]
We probably don't want to use `zookeeper.service.consul`, as the zk client does not refresh DNS entries, so we lose benefits of RR dns. The zk hosts should be enumerated.

**Comment by tymofii-polekhin** [comment]
@stevendborrelli 
> the zk client does not refresh DNS entries
THAT explains a lot... :)

**Comment by ryane** [comment]
kafka will start using the list of zk hosts once #1308 is merged
